client/lib/plugins/store.js

```diff
@@ -248,8 +243,8 @@ PluginsStore = {
 		if ( ! plugins ) {
 			return;
 		}
-		plugins = plugins.filter( _filters.isEqual.bind( this, pluginSlug ) );
-		plugin = plugins.pop();
+
+		plugin = find( plugins, _filters.isEqual.bind( this, pluginSlug ) );
```

Comment:
Yeah, it's hard to tell. I'd guessed on the naming and surrounding logic that it was expected there'd be a maximum of one result from the filter.

cc @enejb Do you happen to recall? (re: 6132-gh-calypso-pre-oss)